python/GangaDirac/Lib/Files/DiracFile.py

```diff
@@ -193,35 +190,38 @@ def locations(self):
 
     def _setLFNnamePattern(self, _lfn="", _namePattern=""):
 
-        ## TODO REPLACE THIS WITH IN LIST OF VONAMES KNOWN
-        # Check for /lhcb/some/path or /gridpp/some/path
-        if _namePattern.split(os.pathsep)[0] == self.defaultSE \
-            or (len(_lfn) > 3 and _lfn[0:4] == "LFN:" and len(_namePattern.split(os.pathsep)) >= 1\
-                    and _namePattern.split(os.pathsep)[1] == self.defaultSE):
-            # Check for LFN:/gridpp/some/path or others...
-            temp = _lfn
-            _lfn = _namePattern
-            _namePattern = temp
-
-        if _lfn != "" and _lfn is not None:
-            if len(_lfn) > 3 and _lfn[0:4] == "LFN:":
-                _lfn = _lfn[4:]
-
-        if _lfn != "" and _namePattern != "":
-            self.lfn = _lfn
-            self.remoteDir = os.path.dirname(_lfn)
-            self.namePattern = os.path.basename(_namePattern)
-            self.localDir = os.path.dirname(expandfilename(_namePattern))
-
-        elif _lfn != "" and _namePattern == "":
-            self.lfn = _lfn
+        lfn = copy.deepcopy(_lfn)
+        namePattern = copy.deepcopy(_namePattern)
```

Comment:
Is `_namePattern` a string? If so then it's immutable and there should be no need to do a deepcopy.